src/cli/StatusLogic.ts

```diff
@@ -0,0 +1,70 @@
+import ContractInteractor, { RelayServerRegistered } from '../relayclient/ContractInteractor'
+import HttpClient from '../relayclient/HttpClient'
+import PingResponse from '../common/PingResponse'
+import { RelayRegisteredEventInfo } from '../relayclient/types/RelayRegisteredEventInfo'
+import { Address } from '../relayclient/types/Aliases'
+
+interface StatusConfig {
+  blockHistoryCount: number
+  getAddressTimeout: number
+  relayHubAddress: Address
+}
+
+interface PingAttempt {
+  pingResponse?: PingResponse
+  error?: Error
+}
+
+interface Statistics {
+  totalStakesByRelays: string
+  relayRegisteredEvents: RelayRegisteredEventInfo[]
+  relayPings: Map<string, PingAttempt>
+  balances: Map<Address, string>
+}
+
+export default class StatusLogic {
+  private readonly contractInteractor: ContractInteractor
+  private readonly httpClient: HttpClient
+  private readonly config: StatusConfig
+
+  constructor (contractInteractor: ContractInteractor, httpClient: HttpClient, config: StatusConfig) {
+    this.contractInteractor = contractInteractor
+    this.httpClient = httpClient
+    this.config = config
+  }
+
+  async gatherStatistics (): Promise<Statistics> {
```

Comment:
I don't think its a good approach to lump everything into one method.
- it can only be used for one purpose - the status cmd
- its too slow (many round-trips) for any other purpose (e.g. gui statistics)
- timeout of any request will break the entire structure, and you'll have to start all over (rare, but does happen)
- no way to do operations in parallel, even those unrelated.

notice that you don't return any info, until the last relay request times out (and I assume that once there are more than 1-2 relays, some relays will be "zombies" and time-out.

Reply:
I did not want to revive the status tool initially, but then decided it was important enough to have it working. I will work on improving the code and performance of it later, at least now it works for 'one relay on ganache' use-case, which is what we target in the ongoing hackathon.

Reply:
Added to backlog.